### PR TITLE
`run_session`

### PR DESCRIPTION
**Pull request by clarisw**

### CODE REVIEW

File: Banyan/src/sessions.jl [hunk 1]
```diff
@@ -342,4 +345,35 @@ function with_session(f::Function; kwargs...)
...
+    url::Union{String,Nothing} = nothing,
+    branch::Union{String,Nothing} = nothing,
+    directory::Union{String,Nothing} = nothing,
+    dev_paths::Union{Vector,Nothing} = [],
+    force_clone::Union{Bool,Nothing} = false,
+    force_pull::Union{Bool,Nothing} = false,
+    force_install::Union{Bool,Nothing} = false,
+    estimate_available_memory::Union{Bool,Nothing} = true,
+    email_when_ready::Union{Bool,Nothing}=nothing,
+    kwargs...,)::SessionId
```
Comment (cailinw) [inline Banyan/src/sessions.jl:382]: Maybe @calebwin knows, but I"m not sure it's needed to list out all the params like this.
I'm wondering whether run_session can just accept kwargs, and then pass that to start_session?

File: Banyan/src/sessions.jl [hunk 2]
```diff
@@ -181,8 +180,12 @@ function start_session(;
...
     session_id = response["session_id"]
     resource_id = response["resource_id"]
-    @info "Starting session with ID $session_id on cluster named \"$cluster_name\""
-
+    if for_running == false
+        @info "Running session with $code_files"    
+    else
+        @info "Starting session with ID $session_id on cluster named \"$cluster_name\""
+    end
+    println("donneee")
```
Comment (calebwin) [inline Banyan/src/sessions.jl]: Remove this

File: Banyan/src/sessions.jl (hunk 1 above)
Comment (calebwin) [reply Banyan/src/sessions.jl:382]: That's probably better but there are some arguments that `start_session` accepts that `run_session` doesn't like `force_update_files`. I think for now this is fine.

File: Banyan/src/sessions.jl [hunk 3]
```diff
@@ -351,4 +354,66 @@ function with_session(f::Function; kwargs...)
...
+    
+
+    force_update_files = true
+    start_session(;cluster_name = cluster_name, nworkers = nworkers, release_resources_after = release_resources_after, 
+                  print_logs = print_logs, store_logs_in_s3 = store_logs_in_s3, store_logs_on_cluster = store_logs_on_cluster, 
+                  sample_rate = sample_rate, session_name = session_name, files = files, code_files = code_files, force_update_files = force_update_files,
+                  pf_dispatch_table = pf_dispatch_table, using_modules = using_modules, url = url, branch = branch,
+                  directory = directory, dev_paths = dev_paths, force_clone = force_clone, force_pull = force_pull, force_install = force_install, 
+                  estimate_available_memory = estimate_available_memory, nowait = false, email_when_ready = email_when_ready, for_running = true)
+    end_session(get_session_id)
```
Comment (calebwin) [inline Banyan/src/sessions.jl]: Make a try-catch-finally (google what that looks like in Julia) and put the `start_session` call in the `try` block; in the `catch` block, put the `end_session` call followed by `rethrow()`, and in the `finally` block put the `end_session` call.

Comment (calebwin) [inline Banyan/src/sessions.jl]: Change to `get_session_id()`

**Review by calebwin — CHANGES_REQUESTED** [review]

**Pull request merged**